src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -913,12 +916,11 @@ def _build_application_creds(password=None, key_value=None, key_type=None,  # py
 
 
 def create_service_principal(identifier):
-    return _create_service_principal(identifier)
-
-
-def _create_service_principal(identifier, resolve_app=True):
     client = _graph_client_factory()
+    return _create_service_principal(client, identifier)
+
 
+def _create_service_principal(client, identifier, resolve_app=True):
```

Comment:
`craete_service_principal` is an overload version of this method. Although python doesn't have the concept of overload method, I think it is more suitable to combine these two methods back together allowed keyworded parameters and determine whether a `client` is passed in or generated by checking if `client` parameter is `None`.